test/integration/src/test/java/io/pravega/test/integration/KVTableWithStreamTest.java

```diff
@@ -0,0 +1,769 @@
+                        RuntimeException::new).getSegments().size());
+
+        assertEquals("The stream segment count should be same as passed", streamSegments,
+                Futures.getAndHandleExceptions(this.controller.getCurrentSegments(newScope, streamName),
+                        RuntimeException::new).getSegments().size());
+
+        try (ConnectionFactory connectionFactory = new SocketConnectionFactoryImpl(ClientConfig.builder().build());
+             ClientFactoryImpl clientFactory = new ClientFactoryImpl(newScope, this.controller, connectionFactory);
+             ReaderGroupManager readerGroupManager = new ReaderGroupManagerImpl(newScope, this.controller, clientFactory)) {
+
+            log.info("Creating writer");
+            @Cleanup
+            final EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName,
+                    new JavaSerializer<>(), EventWriterConfig.builder().build());
+
+            // start writing events to the stream.
+            for (int i = 0; i < numEvents; i++) {
+                log.info("Writing event {}", streamEvent);
+                writer.writeEvent(routingKey, streamEvent);
+                writeCount++;
+                writer.flush();
+            }
+            log.info("Closing writer {}", writer);
+            writer.close();
+
+            // create a reader group.
+            log.info("Creating Reader group : {}", readerGroupName);
+
+            readerGroupManager.createReaderGroup(readerGroupName, ReaderGroupConfig.builder()
+                    .stream(Stream.of(newScope, streamName)).build());
+            log.info("Reader group name {} ", readerGroupManager.getReaderGroup(readerGroupName).getGroupName());
+            log.info("Reader group scope {}", readerGroupManager.getReaderGroup(readerGroupName).getScope());
+
+            log.info("Creating reader");
+            @Cleanup
+            final EventStreamReader<String> reader = clientFactory.createReader(readerName,
+                    readerGroupName,
+                    new JavaSerializer<>(),
+                    ReaderConfig.builder().build());
+
+            // start reading events from the stream.
+            while (!(readCount == writeCount)) {
+                final String eventRead = reader.readNextEvent(SECONDS.toMillis(2)).getEvent();
+                log.info("Reading event {}", eventRead);
+                if (eventRead != null) {
+                    // update if event read is not null.
+                    readCount++;
+                }
+            }
+            log.info("Closing reader {}", reader);
+            reader.close();
+
+            //delete readergroup
+            log.info("Deleting readergroup {}", readerGroupName);
+            readerGroupManager.deleteReaderGroup(readerGroupName);
+        }
+        log.info("All writers have stopped. Event Written Count:{}, Event Read " +
```

Comment:
You can probably fit this on one line.

Reply:
But I doubt its usefulness. You are checking this on the next line so what's the point of logging them if you check them immediately afterwards?

Reply:
Sure, I'll refactor the code here. Thanks